f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -235,20 +241,26 @@ def _populate_member_network(self, context, member, network):
         member['vxlan_vteps'] = []
         member['gre_vteps'] = []
 
-        if 'provider:network_type' in network:
-            net_type = network['provider:network_type']
-            if net_type == 'vxlan':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['vxlan_vteps'] = self._get_endpoints(
-                        context, 'vxlan', host)
-            if net_type == 'gre':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['gre_vteps'] = self._get_endpoints(
-                        context, 'gre', host)
-        if 'provider:network_type' not in network:
-            network['provider:network_type'] = 'undefined'
```

Comment:
Lines 250-251 need to be restored.  I checked your new agent PR and there are multiple unprotected references to this key in assure_bigip_network.  Could cause an attribute error to be raised.